### PR TITLE
refactor: refactor the e2e setup suite

### PR DESCRIPTION
**Pull request by gsk967**
## Description

This pull request aims to refactor the e2e setup test suite, making it easier for us to utilize both QA tests and e2e tests.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: tests/e2e/e2e_test.go [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
+	setup "github.com/umee-network/umee/v4/tests/e2e/setup"
 	"github.com/umee-network/umee/v4/tests/grpc"
 )
 
-func (s *IntegrationTestSuite) TestPhotonTokenTransfers() {
+type E2ETest struct {
+	setup.E2ETestSuite
+}
+
+func TestE2ETesrtSuite(t *testing.T) {
```
Comment (robert-zaremba) [inline tests/e2e/e2e_test.go:19]:
```suggestion
func TestE2ETestSuite(t *testing.T) {
```

File: tests/e2e/setup/chain.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package e2e
+package e2esetup
```
Comment (robert-zaremba) [inline tests/e2e/setup/chain.go:1]: the package name should be `setup`

File: tests/e2e/setup/chain.go [hunk 3]
```diff
@@ -23,7 +23,7 @@ const (
 
 var (
 	encodingConfig sdkparams.EncodingConfig
-	cdc            codec.Codec
+	Cdc            codec.Codec
```
Comment (robert-zaremba) [inline tests/e2e/setup/chain.go:26]: why do you create this as a global? I think it should be part of the `E2E` suite struct.

**Review by robert-zaremba — COMMENTED** [review]
This PR will complicate merge into `no-gb` branch. 
1. Do we need to merge this PR now? If yes then could you handle the next merge `main -> no-gb`?

File: tests/e2e/setup/chain.go (hunk 3 above)
Comment (gsk967) [reply tests/e2e/setup/chain.go]: move the Cdc into E2E test suite

File: tests/e2e/setup/chain.go [hunk 4]
```diff
@@ -38,16 +37,14 @@ func init() {
...
 type chain struct {
 	dataDir        string
-	id             string
-	validators     []*validator
-	orchestrators  []*orchestrator
-	gaiaValidators []*gaiaValidator
+	ID             string
+	Validators     []*validator
+	Orchestrators  []*orchestrator
+	GaiaValidators []*gaiaValidator
```
[the remark this answers is not in the document]
Comment (gsk967) [reply tests/e2e/setup/chain.go:47]: No we need to them in qa_test

**Pull request merged**